Add descriptions to database roles and remove hard limit

mattboehm opened the pull request:
fixes CATALOG-4

tanderegg commented on ckanext/cfpb_extrafields/templates/package/snippets/resource_type-specific/cfpb_extrafields_database.html:
```diff
@@ -47,24 +47,19 @@
-  placeholder="", value=data.db_role_level_9,error=errors.db_role_level_9,classes=['control-full']) }}
+
+  {# Add 20 slots for blank rows to be added #}
+  {% set roles = data.get("db_roles", []) + [("", "")] * 20 %}
```
I think we can start with a smaller number, most datasets will probably only have 2-4 roles, so maybe start with 5 slots?

mattboehm replied:
Basically the code creates 20 slots, but they are all hidden in the webpage. When you click "add role", we make the first hidden set of inputs visible. Therefore, having a big number here doesn't really have any downsides besides adding a few bytes to the page size.

Perhaps I should push this out to the staging instance so that you can take a look there and verify that you are happy with how this looks before we consider merging.

tanderegg replied:
Ah, I get it, sorry I had assumed this worked differently.  Normally I would use JS to insert the fields rather than hide/reveal them, but this should work fine and would be a simpler approach.  Agreed we should push this (plus the PR's I merged) to staging for review, thanks!

mattboehm replied:
@tanderegg I've deployed this code to staging. Let me know if you have any feedback on its behavior or appearance.

tanderegg commented on ckanext/cfpb_extrafields/validators.py:
```diff
@@ -117,3 +117,15 @@ def clean_select_multi(s):
+    items_to_delete = [key for key in data.keys() if key.startswith(ROLE_PREFIX) or key.startswith(DESC_PREFIX)]
+    for key in items_to_delete:
+        del data[key]
+    return data
```
This is a clever way of handling the change in data structure :)  I think ultimately we'll want to do away with the old fixed roles, but it does make sense to do that as a later step.

tanderegg commented on ckanext/cfpb_extrafields/controllers/ldap_search.py:
```diff
@@ -0,0 +1,121 @@
+def get_datasource(source_id):
+    response = get_action("package_show")({}, {"id", source_id})
+    return response
+    # return response["result"]
```
@mattboehm Should we remove this commented out line?

tanderegg commented on ckanext/cfpb_extrafields/controllers/ldap_search.py:
```diff
@@ -0,0 +1,121 @@
+class LdapSearchController(BaseController):
+    def ldap_search(self):
+        """"""
+        # ou = request.params.get("ou")
```
Likewise here, probably should remove this line if no longer needed.

tanderegg commented:
@mattboehm See the Travis job, looks like the --use-mirrors pip flag has been depreceated, could you try removing that from the Travis tests and see if the Travis job completes?  Thanks!

mattboehm commented:
@tanderegg I removed the arg, but it was failing on the `npm install` command. As far as I can tell, we don't run this command when installing to Mesos, so I just removed it for now. If that sounds okay to you, this can be merged and then I can rebase #53 on top of it so that it's ready to be merged.

tanderegg commented:
@mattboehm Yeah, looks like there are a bunch of deprecated node packages, that is probably the problem.  We should probably review all this in more detail at some point, see if we can remove the node stuff completely.  This is good for now though, merging now.
